File: DuckTyping/Python/duckTyping.py
#!/usr/bin/python

class Duck(object):
    def __init__(self, string):
        print('do something ' + string)


    def quack(self):
        print('Quaaaaaaack!')


    def feathers(self):
        print('The duck has white and gray feathers.')


class Person(object):
    def quack(self):
        print('The person imitates a duck.')


    def feathers(self):
        print('The person takes a feather from the ground and shows it.')


def in_the_forest(object):
    object.quack()
    object.feathers()


def game():
    donald = Duck('duck')
    john = Person()
    in_the_forest(donald)
    in_the_forest(john)


if __name__ == "__main__":
    game()

File: DuckTyping/Ruby/duckTyping.rb
class Duck
	def initialize(string)
		@string = string
		puts "do something " + @string
	end

	def quack
		puts "Quaaaaaaack!"
	end

	def feathers
		puts "The duck has white and gray feathers."
	end
end

class Person
	def quack
		puts "The person imitates a duck"
	end

	def feathers
		puts "The person takes a feather from the ground and shows it."
	end
end

def in_the_forest(object)
	object.quack
	object.feathers
end

def game
	donald = Duck.new("duck")
	john = Person.new
	in_the_forest(donald)
	in_the_forest(john)
end

game